drivers/flash/flash_stm32_ospi.c

```diff
@@ -378,18 +381,26 @@ static int stm32_ospi_write_enable(OSPI_HandleTypeDef *hospi, uint8_t nor_mode,
 		return -EIO;
 	}
 
-	/* Configure automatic polling mode to wait for write enabling */
+	/* New command to Configure automatic polling mode to wait for write enabling */
 	if (nor_mode == OSPI_OPI_MODE) {
 		s_command.Instruction = SPI_NOR_OCMD_RDSR;
+		s_command.AddressMode = HAL_OSPI_ADDRESS_8_LINES;
+		s_command.DataMode = HAL_OSPI_DATA_8_LINES;
```

Comment:
`ospi_prepare_cmd` already sets `s_command.AddressMode = HAL_OSPI_ADDRESS_8_LINES` and `s_command.DataMode = HAL_OSPI_DATA_8_LINES`  for `OSPI_OPI_MODE`.

Reply:
yes for the WREN command but here is the RDSR command to be prepared,